### PR TITLE
Introduce NNI with distributed Adaptive API

### PR DESCRIPTION
**Pull request by ZeyaWang**
This PR adds an example to Texar-Pytorch about how to run NNI experiments with distributed Adaptive API with the help of AdaptDL.

`examples/bert/bert_classifier_adaptive_nni.py` is th the nni version of `examples/bert/bert_classifier_adaptive.py`.  `config_bert_classifier.yml` is the configuration file to run the experiment.

### CODE REVIEW

**Comment by ZeyaWang** [comment]
For the internal test, please add this to requirement.txt when building the docker images.

```
--extra-index-url http://pypi.int.petuum.com:8080/simple
--trusted-host pypi.int.petuum.com
nni==1.8.24+petuum
```

File: requirements.txt
Comment (hunterhector) [inline requirements.txt:7]: We can add the requirement of Petuum-NNI when it is done. We can approve this PR but wait for the NNI released to merge

File: examples/bert/README.md
Comment (hunterhector) [inline examples/bert/README.md:276]:
```suggestion
docker build -t registry.petuum.com/dev/nni:texar-bert-classifier -f docker/Dockerfile .
```
I think we need another argument for the path.

File: examples/bert/README.md
Comment (hunterhector) [inline examples/bert/README.md:282]: `provide the command line with the code path in the docker image`
This line is slightly hard to understand, could provide an example.

AdaptDL cluster and NNI setup is better explained at the beginning of this documentation session.

**Review by hunterhector — COMMENTED** [review]
We can wait and merge this only when all dependencies are open-sourced. 

The docker build command here requires a petuum registry, any way for a non-petuum user to reproduce this?

File: examples/bert/README.md
Comment (ZeyaWang) [reply examples/bert/README.md]: Have updated the ReadMe file and given an example for this sentence. The NNI setup has already been given at the beginning of the section of NNI with Texar (See: Hyperparamter tuning with Neural Network Intelligence (NNI)). I also added one sentence for AdaptDL setup at the beginning of the section of AdaptDL, and I also suggest the author of that section could add more information about the setup.

File: docker/Dockerfile
Comment (ZeyaWang) [reply docker/Dockerfile:19]: Yeah, for the current setup of NNI, pytorch:1.4 is needed. It won't affect adaptdl-texar in my test but I am not sure if it will affect other examples.

File: requirements.txt
Comment (ZeyaWang) [reply requirements.txt:7]: Sure, will work on it after Petuum-NNI is released.

Comment (ZeyaWang) [reply requirements.txt:9]: The latest NNI version has been updated with adl training service, so it can be directly installed for our nni-adl example. I have update the requirement.txt to include this dependency.

**Comment by ZeyaWang** [comment]
> We can wait and merge this only when all dependencies are open-sourced.
> 
> The docker build command here requires a petuum registry, any way for a non-petuum user to reproduce this?

In the updated ReadMe file, we further provide guidance about how to build and push a docker image to their own private registry and a link about how to create Secrete and use Secrete to pull image from the private registry in K8S. With this guide, the user is able to reproduce all the steps about the docker images.

**Review by hunterhector — APPROVED** [review]

**Pull request merged**
